be/src/storage/txn_manager.cpp

```diff
@@ -269,12 +289,24 @@ Status TxnManager::persist_tablet_related_txns(const std::vector<TabletSharedPtr
 void TxnManager::flush_dirs(std::unordered_set<DataDir*>& affected_dirs) {
     int64_t duration_ns = 0;
     SCOPED_RAW_TIMER(&duration_ns);
+
+    std::vector<std::pair<Status, std::string>> pair_vec;
+    pair_vec.reserve(affected_dirs.size());
+    auto token = _thread_pool_flush->new_token(ThreadPool::ExecutionMode::CONCURRENT);
     for (auto dir : affected_dirs) {
-        auto st = dir->get_meta()->flush();
-        if (!st.ok()) {
-            LOG(WARNING) << "failed to flush tablet meta, dir:" << dir->path() << " res:" << st;
+        Status st;
+        token->submit_func([&]() { st = std::move(dir->get_meta()->flush()); });
+        pair_vec.push_back(std::make_pair(st, dir->path()));
```

Comment:
looks like `st` is not thread safe, `st` may get invalided

Reply:
Status's move and copy constructor is properly implemented, i think there is no thread safe problem, plz give a detailed description.

Reply:
Does the std::move() can guarantee the atomic operation?
```
    Status& operator=(Status&& s) noexcept {
        if (this != &s) {
            Status tmp(std::move(s));
            std::swap(this->_state, tmp._state);
        }
        return *this;
    }
```